src/lib/ares_process.c

```diff
@@ -886,12 +905,18 @@ static ares_status_t process_answer(ares_channel_t      *channel,
     is_cached = ARES_TRUE;
   }
 
-  server_set_good(server, query->using_tcp);
+  server_increment_successes(server, query->using_tcp);
   end_query(channel, server, query, ARES_SUCCESS, rdnsrec);
 
   status = ARES_SUCCESS;
 
 cleanup:
+  /* If we were probing for the server to come back online, lets mark it as
+   * no longer being probed */
+  if (server != NULL) {
+    server->probe_pending = ARES_FALSE;
```

Comment:
this is done universally by end_query(), was there an instance you found this not being called?  What was the call path for that?